### PR TITLE
Vite migration guide

### PR DESCRIPTION
**Pull request by Julien-R44**
Just a draft for now to get some initial feedback. I still have some things to add. I'll try to migrate some repo to Vite to see if I miss some elements

### CODE REVIEW

File: content/cookbooks/migrating-to-vite.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+---
+datetime: 2022-11-01
+author: Julien Ripouteau
+avatarUrl: https://avatars.githubusercontent.com/u/8337858?v=4
+summary: Migrating from Webpack Encore to Vite
+---
+
+This guide will help you to migrate your AdonisJS application from Webpack Encore to Vite.
+
+If you encounter a problem or a missing step in the migration guide, feel free to open an issue or propose a PR as it is more than likely that not all cases are covered.
```
Comment (RomainLanz) [inline content/cookbooks/migrating-to-vite.md]:
```suggestion
If you encounter a problem or a missing step in the migration guide, feel free to open an issue or propose a PR, as it is more than likely that only some cases are covered.
```

File: content/cookbooks/migrating-to-vite.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+```sh
+node ace configure vite
+```
+
+This command will :
+- Create a basic `vite.config.ts` file in the root of your project.
+- Create a `resources/js/app.ts` file. This file will be the entry point of your application.
+- Install the required dependencies : 
+  - `vite` Obviously.
+  - `@adonisjs/vite-plugin-adonis`: Plugin for vite that bridges the gap between AdonisJS and Vite.
```
Comment (RomainLanz) [inline content/cookbooks/migrating-to-vite.md]:
```suggestion
  - `vite`
  - `@adonisjs/vite-plugin-adonis`: Plugin for vite bridging the gap between AdonisJS and Vite.
```

File: content/cookbooks/migrating-to-vite.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+// title: .adonisrc.json
+{
+  "assetsDriver": "vite"
+  // ...
+}
+```
+
+## Vite configuration
+
+You should have a now `vite.config.ts` file at the root of your project with the following content :
```
Comment (RomainLanz) [inline content/cookbooks/migrating-to-vite.md]:
```suggestion
You should have a new `vite.config.ts` file at the root of your project with the following content : 
```

File: content/cookbooks/migrating-to-vite.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    Adonis({
+      entryPoints: {
+        app: ['resources/js/app.ts', 'resources/css/app.css'],
+      },
+    }),
+  ],
+})
+```
+
+Similar to Webpack Encore, you can define multiple entry points. Here, for the sake of simplicity, we only have one entry point named `app`. This entry point can be used in your views by using the well-known `@entryPointsStyles('app)` and `@entryPointsScripts('app')` helpers tags.
```
Comment (RomainLanz) [inline content/cookbooks/migrating-to-vite.md]:
```suggestion
Similar to Webpack Encore, you can define multiple entry points. Here, for the sake of simplicity, we only have one entry point named `app`. This entry point can be used in your views using the well-known `@entryPointsStyles('app)` and `@entryPointsScripts('app')` helpers tags.
```

File: content/cookbooks/migrating-to-vite.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  @vite()
+  // highlight-end
+  @entryPointStyles('app')
+  @entryPointScripts('app')
+</head>
+```
+
+## Typescript, CSS, Tailwind ..
+
+Typescript, Css, Postcss, Less, Sass, Tailwind : these tools should just work out of the box. You don't need to configure anything thanks to Vite ⚡.
```
Comment (RomainLanz) [inline content/cookbooks/migrating-to-vite.md]:
```suggestion
Typescript, CSS, Postcss, Less, Sass, Tailwind: these tools should work out of the box. You don't need to configure anything thanks to Vite ⚡.
```

File: content/cookbooks/migrating-to-vite.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+})
+```
+
+Similar to Webpack Encore, you can define multiple entry points. Here, for the sake of simplicity, we only have one entry point named `app`. This entry point can be used in your views by using the well-known `@entryPointsStyles('app)` and `@entryPointsScripts('app')` helpers tags.
+
+So make sure to add all your `Encore.addEntry()` calls to your new Vite configuration.
+
+## Vite compatible imports
+
+Vite only supports ES modules, so you will need to replace any `require()` statements with `import`.
```
Comment (RomainLanz) [inline content/cookbooks/migrating-to-vite.md:99]: Maybe we should add that at the top of this page (and the Asset Manager doc) with a "warning" box?

**Review by RomainLanz — CHANGES_REQUESTED** [review]

**Comment by Julien-R44** [comment]
I think we're pretty good with this. We should collect feedback as people migrate to see if we are missing some things

Just a little reminder: We still have to update the version numbers at the top of the document, once the PRs are accepted and published

**Comment by Julien-R44** [comment]
Closing since the implementation has changed and therefore guide is now outdated

**Pull request closed without merging**